src/Orchard.Web/Core/Contents/Drivers/ContentsDriver.cs

```diff
@@ -21,8 +21,16 @@ public class ContentsDriver : ContentPartDriver<ContentPart> {
         protected override DriverResult Editor(ContentPart part, dynamic shapeHelper) {
             var results = new List<DriverResult> { ContentShape("Content_SaveButton", saveButton => saveButton) };
 
-            if (part.TypeDefinition.Settings.GetModel<ContentTypeSettings>().Draftable)
+            if (part.TypeDefinition.Settings.GetModel<ContentTypeSettings>().Draftable) {
                 results.Add(ContentShape("Content_PublishButton", publishButton => publishButton));
+                if (part.ContentItem.IsPublished()) {
+                    results.Add(ContentShape("Content_UnpublishButton", unpublishButton => unpublishButton));
+                }
```

Comment:
Re: [the previous comment](https://github.com/OrchardCMS/Orchard/pull/8566#discussion_r899343079) - shouldn't be just this then, since the template already checks `IsPublished`?
```suggestion
                results.Add(ContentShape("Content_UnpublishButton", unpublishButton => unpublishButton));
```

Reply:
I don't remember right now. I'll check asap.

Reply:
The `Content.UnpublishButton` template _does_ check if the item is published.